packages/backend/src/models/CatalogModel/CatalogModel.ts

```diff
@@ -891,13 +893,22 @@ export class CatalogModel {
                 catalogSize: paginatedCatalogItems.data.length,
             },
             async () =>
-                paginatedCatalogItems.data.map((item) =>
-                    parseCatalog({
+                paginatedCatalogItems.data.map((item) => {
+                    let { explore } = item;
+                    if (changeset) {
+                        const exploreWithChanges =
+                            ChangesetUtils.applyChangeset(changeset, {
+                                [item.explore.name]: item.explore,
+                            })[item.explore.name] as Explore; // at this point we know the explore is valid
+                        explore = exploreWithChanges;
+                    }
+                    return parseCatalog({
                         ...item,
+                        explore,
                         catalog_tags:
                             tagsPerItem[item.catalog_search_uuid] ?? [],
-                    }),
-                ),
+                    });
+                }),
```

Comment:
do you think I should wrap in a sentry transaction? 

I don't like the fact that each item has an explore object, but the util to apply a changeset must be super fast